data/mods/ssb/abilities.js

```diff
@@ -815,6 +815,38 @@ let BattleAbilities = {
 			if (move.type === 'Flying') move.accuracy = true;
 		},
 	},
+	// Ransei
+	superguarda: {
+		desc: "This user's Attack is doubled until it is hit by a super effective attack. If this Pokemon is statused, its Attack is 1.5x; ignores burn halving physical damage. This Pokemon can only be damaged by direct attacks.",
+		shortDesc: "Atk 2x until hit by SE move. 1.5x Atk if statused. Immune to indirect dmg.",
+		id: "superguarda",
+		name: "Superguarda",
+		isNonstandard: "Custom",
+		onDamage(damage, target, source, effect) {
+			if (effect.effectType !== 'Move') {
+				return false;
+			}
+		},
+		onAfterDamage(damage, target, source, move) {
+			if (target.getMoveHitData(move).typeMod > 0) {
+				if (!target.heavilydamaged) {
+					this.add('-message', `${target.name}'s attack was reduced after that super effective attack!`);
+					target.heavilydamaged = true;
+				}
+				this.add(`c|@Ransei|Yo really? Why do you keep hitting me with super effective moves?`);
+			}
+		},
+		onModifyAtk(atk, pokemon) {
```

Comment:
I am not sure if i did the correctly due to how the abilities interact with each other

Reply:
I think this is ok.